Assets/Scripts/Input/DragonPlayerController.cs

```diff
@@ -50,6 +52,9 @@ protected override void OnEnable()
         if (photonView.IsMine)
         {
             Combat.Health.UpdateHealthEvent.AddListener(healthUI.UpdateDragonHealthUI);
+            Combat.Resource.UpdateResourceEvent.AddListener(manaUI.UpdateManaUI);
+            Combat.Resource.RegenerateResourceEvent.AddListener(manaUI.RegenerateManaUI);
+            Combat.Resource.StopRegenResourceEvent.AddListener(manaUI.StopRegenManaUI);
```

Comment:
When adding listeners to an event in `OnEnable`, it's advisable to remove them in `OnDisable` as it can result in a memory leak otherwise.

Reply:
👍